vyxal/helpers.py

```diff
@@ -165,9 +165,6 @@ def foldl(
     ctx: Context,
 ) -> Any:
     """Reduce vector by function"""
-    if len(vector) == 0:
-        return 0
```

Comment:
This breaks other elements that use foldl.

Reply:
It doesn't, notice the `initial=None`, and if `initial` is `None`, it defaults to `0`. (I don't know why user didn't just do `initial=0`).

Reply:
The fault is in those other elements, not `foldl`. They should give an initial value or check for empty lists themselves.